src/java.base/share/classes/jdk/internal/classfile/impl/ClassReaderImpl.java

```diff
@@ -399,13 +399,16 @@ public AbstractPoolEntry.Utf8EntryImpl utf8EntryByIndex(int index) {
             int offset = cpOffset[index];
             int tag = readU1(offset);
             final int q = offset + 1;
-            if (tag != TAG_UTF8) throw new ConstantPoolException("Not a UTF8 - index: " + index);
-            AbstractPoolEntry.Utf8EntryImpl uinfo
-                    = new AbstractPoolEntry.Utf8EntryImpl(this, index, buffer, q + 2, readU2(q));
-            cp[index] = uinfo;
-            return uinfo;
+            if (tag == TAG_UTF8) {
```

Comment:
Can we call into the tag accepting entryByIndex? e.g.,
```
if (entryByIndex(index, TAG_UTF8) instanceof AbstractPoolEntry.Utf8EntryImpl utf8) {
  return ...
}
throw new ...
```
?
```

Reply:
Yes, it is a good opportunity to reduce the code a bit.
Thanks!